components/Modal/Modal.tsx

```diff
@@ -45,6 +45,21 @@ export const Modal: React.FC<IModalComponentProps> = withMooskinContext((props)
 		}
 
 		return React.Children.map(children, (child, i) => {
+			// on press esc button close modal
+			const escButtonHandler = (event:any) => {				
+				if(event.keyCode === 27){
+					if (React.isValidElement<IModalOverlayComponentProps>(child) && child.type === ModalCloseButton) {
+						batchClickHandler(event, child.props.onClick)
```

Comment:
No need to call batch click handler, this is used if an on click handler is passed down to either the close button or the overlay, in this case, calling props.onClose is enough.